source/client/stream_decoder.cc

```diff
@@ -37,9 +37,8 @@ void StreamDecoder::onComplete(bool success) {
   }
   ASSERT(!success || complete_);
   decoder_completion_callback_.onComplete(success, *response_headers_);
-  if (success) {
-    caller_completion_callback_();
-  }
+  // XXX(oschaaf): forward success state
```

Comment:
It this hard to do now?

Reply:
Not hard, but it did bloat this PR for a bit.
Added in https://github.com/envoyproxy/nighthawk/pull/130/commits/bd4704f4252471f30cc37e3e9316de47b6a085b2
Let me know if you prefer that backed out of this PR; I can create a separate PR for that too.
(I didn't do it earlier, because I was trying to keep PR size down, so I drew an arbitrary line)

Reply:
I am find with doing it here.  Thanx!